x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,314 @@
+				EndTime:           awssdk.ToTime(deEvents[i].EndTime),
+				EventScopeCode:    awssdk.ToString((*string)(&deEvents[i].EventScopeCode)),
+				EventTypeCategory: awssdk.ToString((*string)(&deEvents[i].EventTypeCategory)),
+				EventTypeCode:     awssdk.ToString(deEvents[i].EventTypeCode),
+				LastUpdatedTime:   awssdk.ToTime(deEvents[i].LastUpdatedTime),
+				Region:            awssdk.ToString(deEvents[i].Region),
+				Service:           awssdk.ToString(deEvents[i].Service),
+				StartTime:         awssdk.ToTime(deEvents[i].StartTime),
+				StatusCode:        awssdk.ToString((*string)(&deEvents[i].StatusCode)),
+			})
+			eventArns[i] = awssdk.ToString(deEvents[i].Arn)
+		}
+		// Fetch event details for the current page of events
+		eventDetails, err := awsHealth.DescribeEventDetails(ctx, &health.DescribeEventDetailsInput{
+			EventArns: eventArns,
+			Locale:    &locale,
+		})
+		if err != nil {
+			m.Logger().Errorf("[AWS Health] DescribeEventDetails failed with : %w", err)
+			break
+		}
+		// Fetch event description for the current page of events
+		successSet := eventDetails.SuccessfulSet
+		for x := range successSet {
+			for y := range healthDetailsTemp {
+				if awssdk.ToString(successSet[x].Event.Arn) == healthDetailsTemp[y].EventArn {
+					healthDetailsTemp[y].EventDescription = awssdk.ToString(successSet[x].EventDescription.LatestDescription)
+				}
+			}
+		}
+		// Fetch affected entities related to event ARNs in the current page
+		affInputParams = health.DescribeAffectedEntitiesInput{
+			Filter: &types.EntityFilter{
+				EventArns: eventArns,
+			},
+		}
+		affPage = *health.NewDescribeAffectedEntitiesPaginator(
+			awsHealth,
+			&affInputParams,
+			affOptFn,
+		)
+
+		for affPage.HasMorePages() {
+			// Fetch current page of affected entities
+			affCurrentPage, err := affPage.NextPage(ctx)
+			if err != nil {
+				m.Logger().Errorf("[AWS Health] DescribeAffectedEntitie failed with : %w", err)
+				break
+			}
+			// Extract relevant details of affected entities and match them with event details
+			for k := range affCurrentPage.Entities {
+				affEntityTemp = AffectedEntityDetails{
+					AwsAccountId:    awssdk.ToString(affCurrentPage.Entities[k].AwsAccountId),
+					EntityUrl:       awssdk.ToString(affCurrentPage.Entities[k].EntityUrl),
+					EntityValue:     awssdk.ToString(affCurrentPage.Entities[k].EntityValue),
+					LastUpdatedTime: awssdk.ToTime(affCurrentPage.Entities[k].LastUpdatedTime),
+					StatusCode:      awssdk.ToString((*string)(&affCurrentPage.Entities[k].StatusCode)),
+					EntityArn:       awssdk.ToString(affCurrentPage.Entities[k].EntityArn),
+				}
+				for l := range healthDetailsTemp {
+
+					if awssdk.ToString(affCurrentPage.Entities[k].EventArn) == healthDetailsTemp[l].EventArn {
+						healthDetailsTemp[l].AffectedEntities = append(healthDetailsTemp[l].AffectedEntities, affEntityTemp)
+						switch awssdk.ToString((*string)(&affCurrentPage.Entities[k].StatusCode)) {
```

Comment:
is there a reason why the enums in the SDK do not contain values for `PENDING` and `RESOLVED`? there seems to be ambiguity in the docs on this matter as well.